### PR TITLE
CI fixes

### PR DESCRIPTION
**Pull request by TriplEight**
- running 
```
cargo doc --no-deps --all-features -p type-metadata -p ink_abi -p ink_abi_derive -p ink_core -p ink_core_derive -p ink_primitives -p ink_prelude -p ink_lang -p ink_lang_macro
```
to cover all packages.
- also set `GIT_DEPTH` to 100 to avoid some potential failures I've had with substrate, it will not affect CI speed more than on a second.
- had to `cargo upgrade` to satisfy `trybuild v1.0.23` which now expects everything to be up-to-date. (and [the pipeline](https://gitlab.parity.io/parity/ink/pipelines/82198) succeded)

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -212,10 +212,10 @@ publish-docs:
...
   variables:
     GIT_DEPTH:                     0
-  only:
-    - master
-    - schedules
-    - tags
+  # only:
+  #   - master
+  #   - schedules
+  #   - tags
```
Comment (athei) [inline .gitlab-ci.yml]: Please delete code that should be removed instead of commenting it.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ stages:
 
 variables:
   GIT_STRATEGY:                    fetch
-  GIT_DEPTH:                       3
+  GIT_DEPTH:                       100
```
Comment (athei) [inline .gitlab-ci.yml]: Why is this increased to such a huge number? Also it seems unrelated to the PR.

Comment (TriplEight) [reply .gitlab-ci.yml]: also checking one theory, will place WIP here to solve some issues

File: .gitlab-ci.yml (hunk 1 above)
Comment (TriplEight) [reply .gitlab-ci.yml]: it's commented for testing purposes, shouldn't be removed

**Comment by TriplEight** [comment]
I too do not like it that much, but it's not that bad since it's not strict bound to exact versions (as with `"=1.0.23"`. Besides, after fixing that bug in `trybuild` we will be able to return to the more convenient versioning.
